shared/test/test_assets.rb

```diff
@@ -454,4 +446,21 @@ def post_file(assets, channel_id, filename, contents, content_type)
     response = post(assets, channel_id, file)
     [response, tmp_filename]
   end
+
+  def assert_custom_metric(index, metric_type, length_msg = nil)
+    # Filter out metrics from other test cases.
+    metrics = NewRelic::Agent.get_metrics %r{^Custom/FilesApi}
+    length_msg ||= "custom metrics recorded: #{index}"
+    assert_equal index, metrics.length, length_msg
+    last_metric = metrics.last
```

Comment:
This method verifies that the last metric added was the nth, and that it has the properties specified below. The assumption is that you will have checked the properties of other metrics earlier if you care about them.